codec-dns/src/test/java/io/netty/handler/codec/dns/DefaultDnsRecordEncoderTest.java

```diff
@@ -63,4 +66,83 @@ private static void testEncodeName(byte[] expected, String name) throws Exceptio
             expectedBuf.release();
         }
     }
+
+    @Test
+    public void testIpv4() throws Exception {
+        testIp(InetAddress.getByName("1.2.3.4"));
+    }
+
+    @Test
+    public void testIpv6() throws Exception {
+        testIp(InetAddress.getByName("::0"));
+    }
+
+    private static void testIp(InetAddress address) throws Exception {
+        int addressBits = address.getAddress().length * Byte.SIZE;
+        for (int i = 0; i <= addressBits; i += 1) {
+            testIp(address, i);
+        }
+    }
+
+    private static void testIp(InetAddress address, int prefix) throws Exception {
+        ByteBuf addressPart = Unpooled.wrappedBuffer(address.getAddress(), 0,
+                DefaultDnsRecordEncoder.calculateEcsAddressLength(prefix));
```

Comment:
we are already doing half of the computation below ... i'm not sure it is worth sharing `calculateEcsAddressLength` with this unit test if there needs to be some duplication anyways.